### ISSUE TITLE
RStudio loads xfun and yaml packages for no reason

### ISSUE THREAD

**Issue by arencambre** [rstudio/rstudio#8698]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.4.1102
    OS Version      : Windows 10 20H2
    R Version       : 4.0.3

### Steps to reproduce the problem
Open RStudio and do nothing else. In my case, a project is not loaded, but an unsaved R script and a view of an object named `x` come up since RStudio restores my prior state.

### Describe the problem in detail
When I update packages, I am prompted to restart R:
![image](https://user-images.githubusercontent.com/411495/103903289-5468da00-50c1-11eb-9842-8d53e6a2429d.png)

That happens because one of the packages ready for an update is `xfun`, and rsession.exe has a lock on xfun.dll (and yaml.dll):
![image](https://user-images.githubusercontent.com/411495/103904183-8cbce800-50c2-11eb-86a4-98bd50ed21e7.png)

This is despite that RStudio says neither package is loaded:
![image](https://user-images.githubusercontent.com/411495/103904905-87ac6880-50c3-11eb-8909-6a7209f6fbb2.png)

Reviewing [xfun's documentation](https://cran.r-project.org/web/packages/xfun/index.html), the **Reverse imports** section suggests it's used mainly with libraries associated with Rmds and also possibly with other libraries.

The problem is I have no Rmds open:
![image](https://user-images.githubusercontent.com/411495/103904362-caba0c00-50c2-11eb-8c0a-5ca305cffca5.png)

Pressing **Yes** on that dialog caused a successful update of `xfun`. However, after closing and reopening RStudio, it still puts a lock on `xfun.dll` and `yaml.dll`.

### Describe the behavior you expected
RStudio does not cause rsession.exe to load libraries that are needed for functionality I am not using.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting a RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by arencambre** [comment]
Something similar is happening on a different PC:
![image](https://user-images.githubusercontent.com/411495/103935895-aecb6000-50ec-11eb-8972-850cef3e3c35.png)

**Comment by kevinushey** [comment]
If you want to find out why these packages are being loaded, you can put the following in your `~/.Rprofile`:

```
trace(loadNamespace, quote({
  print(package)
  print(rlang::trace_back())
}))
```

Then share the traceback you see when `xfun` is loaded.

**Comment by arencambre** [comment]
Thank you. While that shared more information, it is still inconclusive.

With that .Rprofile, these are the outputs of starting R directly and also starting it through RStudio:

- [just R.txt](https://github.com/rstudio/rstudio/files/5783909/just.R.txt)
- [rstudio.txt](https://github.com/rstudio/rstudio/files/5783910/rstudio.txt)

The only substantive differences between them are:

- `[Previously saved workspace restored]` appears on line 21 in the "just R" file and not in the "RStudio" one
- the "RStudio" one has many more packages being loaded for no apparent reason

**Comment by kevinushey** [comment]
Thanks; I can reproduce. The log helps us narrow it down:

```
[1] "tinytex"
     x
  1. \-(function () ...
  2.   +-base::tryCatch(tinytex:::tinytex_root(), error = function(e) .rs.tinytexRoot())
  3.   | \-base:::tryCatchList(expr, classes, parentenv, handlers)
  4.   |   \-base:::tryCatchOne(expr, names, parentenv, handlers[[1L]])
  5.   |     \-base:::doTryCatch(return(expr), name, parentenv, handler)
  6.   \-tinytex:::tinytex_root
  7.     +-base::get(name, envir = asNamespace(pkg), inherits = FALSE)
  8.     \-base::asNamespace(pkg)
  9.       \-base::getNamespace(ns)
 10.         \-base::loadNamespace(name)
```

And that points to code here:

https://github.com/rstudio/rstudio/blob/d3b104505268387a80a61c09a70b6e1e210e8e0b/src/cpp/session/modules/SessionRMarkdown.R#L549-L552

So, it looks like we're over-aggressively loading `tinytex` through this code path for some reason.

**Comment by kevinushey** [comment]
The `yaml` package is also loaded for a similar reason (enumerating tutorials for view in the Tutorial pane).

**Comment by R-Hannibal** [comment]
Verified Fixed on: RStudio Version - 1.4.1591, R Version - 4.0.3, Windows 10 - Desktop

Tested that the `tinytex` and `yaml` packages are no longer being auto-loaded by installing those packages > opening the tutorial pane > restarting the R Session > running `loadedNamespaces()`. Post-fix output below:

```
> loadedNamespaces()
[1] "compiler"  "graphics"  "tools"     "utils"     "grDevices" "stats"
[7] "datasets"  "methods"   "base"
```
